### PR TITLE
[v16] Re-open the log file by filesystem notify

### PR DESCRIPTION
**Pull request by vapopov**
Backport https://github.com/gravitational/teleport/pull/43359 to branch/v16

changelog: Re-open log file by filesystem notify rename/remove (useful for logrotate).

### CODE REVIEW

**Review by espadolini — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

**Comment by r0mant** [comment]
@vapopov Let's hold this backport until 16.2.

File: lib/utils/log/file_writer.go
Comment (vapopov) [reply lib/utils/log/file_writer.go:64]: added https://github.com/gravitational/teleport/pull/44464, will cherry-pick after merge

**Pull request merged**
